scm-record/src/ui.rs

```diff
@@ -3413,4 +3411,17 @@ mod tests {
         let recorder = Recorder::new(state.clone(), &mut input);
         assert_eq!(recorder.run().unwrap(), state);
     }
+
+    proptest::proptest! {
```

Comment:
cc @firestack this test didn't find any bugs, but maybe you think property-based testing is cool 🙂

Reply:
Oooh very neat! Thanks for sharing, I've been very curious about using `proptest` and doing property testing, but I've not actually worked with it firsthand yet!